doc/manual/typing.md

```diff
@@ -255,6 +255,26 @@ The following type constructors are available:
   }
   ```
 
+### Subtyping
+
+Subtyping is a relation between two types where
+you can use one of the type in place of the other one.
+In Nickel, there is :
```

Comment:
The beginning is a bit dry. The best way to introduce a notion is to motivate it, with an example. For example, start by saying something like: 

> While distinct types are usually incompatible, some types might actually be safely converted to some other types. Take the following example:
>
>```nickel
>let extended : {_ : Number} =
>  let initial = {foo = 1} in
>  std.record.insert initial
>```

And then go on to explain that technically the types don't match, but it's legal to "forget" some information to go from a record type to dictionary type.

And go on to explain why this is accepted while the types don't really match specifically.